createcoveragereport.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# This script merges all coverage files found under the 'coverage' directory
+# and then creates a dotcover XML report and from that creates a browsable
+# ReportGenerator html report.
+
+# Use an appropriate version of nuget... preferring
+# first an existing NUGET variable, then NuGet, then
+# just falling back to the path.
+if [ -z "$NUGET" ]
+then
+  if [ -n "$NuGet" ]
+  then
+    NUGET="$NuGet"
+  else
+    NUGET="nuget"
+  fi
+fi
+set -e
+
+$NUGET install -Verbosity quiet -OutputDirectory packages -Version 2017.1.20170613.162720 JetBrains.dotCover.CommandLineTools
+$NUGET install -Verbosity quiet -OutputDirectory packages -Version 2.4.5.0 ReportGenerator
+
+DOTCOVER=$PWD/packages/JetBrains.dotCover.CommandLineTools.2017.1.20170613.162720/tools/dotCover.exe
+REPORTGENERATOR=$PWD/packages/ReportGenerator.2.4.5.0/tools/ReportGenerator.exe
+FIND=/usr/bin/find
+
+if [ ! -d coverage ]
+then
+ mkdir coverage
+fi
+
+echo "Merging reports..."
+
+echo "<MergeParams>" > "coverage/merge.xml"
+for filename in coverage/*.dvcr; do
+  echo "<Source>$(basename $filename)</Source>" >> "coverage/merge.xml"
+done
+echo "<Output>all.dvcr</Output>" >> "coverage/merge.xml"
+echo "</MergeParams>" >> "coverage/merge.xml"
+
+$DOTCOVER merge coverage/merge.xml
+
+echo "Generating detailed xml report..."
+
+echo "<ReportParams>" > "coverage/report.xml"
+echo "<Source>all.dvcr</Source>" >> "coverage/report.xml"
+echo "<Output>coverage.xml</Output>" >> "coverage/report.xml"
+echo "<ReportType>DetailedXML</ReportType>" >> "coverage/report.xml"
+echo "</ReportParams>" >> "coverage/report.xml"
+
+$DOTCOVER report coverage/report.xml
+
+echo "Running ReportGenerator to create an html report..."
+
+$REPORTGENERATOR \
```

Comment:
I've just looked at version 2.5.10 of ReportGenerator, and that contains file filters - so we should be able to use our previous "find the generated files" code (no doubt with tweaks for how the filter is expressed) and pass it in there. Yay!

Alternatively, before finding that (grr) I wrote a short tool to remove generated files from the XML file. That may be cleaner, given that the complete set of generated files (after merging) would be huge.

I suggest we get this change merged and then I'll add the tool in another commit.